### PR TITLE
Fix: Handle correctly keys on Get/Put Value and Get/Put IPNS.

### PR DESCRIPTION
**Pull request by ajnavarro**

### CODE REVIEW

File: client/store.go [hunk 1]
```diff
@@ -4,18 +4,34 @@ import (
...
 
 // PutValue adds value corresponding to given Key.
 func (c *Client) PutValue(ctx context.Context, key string, val []byte, opts ...routing.Option) error {
-	return c.PutIPNS(ctx, []byte(key), val)
+	_, path, err := record.SplitKey(key)
+	if err != nil {
+		return err
+	}
+
+	return c.PutIPNS(ctx, []byte(path), val)
```
Comment (aschmahmann) [inline client/store.go:19]: We should have a check for if the key is a valid IPNS identifier as in https://github.com/ipfs/go-ipns/blob/f6e884731b943682a432e4719696c21d8f354b9c/record.go#L48-L49.

Whether this lives in the valuestore wrapper or lower in the client seems debatable, either way is fine by me. You can reuse the error if you want or not, doesn't seem a huge deal either way.

**Review by aschmahmann — CHANGES_REQUESTED** [review]

**Comment by BigLep** [comment]
2022-07-07 conversation: in review, focus on making sure we're handling strings properly.

File: client/putipns.go [hunk 2]
```diff
@@ -4,10 +4,17 @@ import (
 	"context"
 
 	"github.com/ipfs/go-delegated-routing/gen/proto"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/pkg/errors"
 )
 
 func (fp *Client) PutIPNS(ctx context.Context, id []byte, record []byte) error {
-	_, err := fp.client.PutIPNS(ctx, &proto.PutIPNSRequest{ID: id, Record: record})
+	_, err := peer.IDFromString(string(id))
```
Comment (aschmahmann) [inline client/putipns.go:12]: I know I pointed you at go-ipns, but the go-libp2p folks want to kill this function anyway  everyone else misusing it https://github.com/libp2p/go-libp2p-core/pull/274.

We can do `IDFromBytes(id)` instead though

File: client/putipns.go [hunk 3]
```diff
@@ -2,12 +2,19 @@ package client
...
 
 	"github.com/ipfs/go-delegated-routing/gen/proto"
+	"github.com/libp2p/go-libp2p-core/peer"
 )
 
 func (fp *Client) PutIPNS(ctx context.Context, id []byte, record []byte) error {
-	_, err := fp.client.PutIPNS(ctx, &proto.PutIPNSRequest{ID: id, Record: record})
+	_, err := peer.IDFromBytes(id)
+	if err != nil {
+		return fmt.Errorf("invalid peer ID. Error: [%w]", err)
```
Comment (Jorropo) [inline client/putipns.go:14]:
```suggestion
		return fmt.Errorf("invalid peer ID: %w", err)
```

File: client/putipns.go [hunk 4]
```diff
@@ -19,6 +26,11 @@ type PutIPNSAsyncResult struct {
 }
 
 func (fp *Client) PutIPNSAsync(ctx context.Context, id []byte, record []byte) (<-chan PutIPNSAsyncResult, error) {
+	_, err := peer.IDFromBytes(id)
+	if err != nil {
+		return nil, fmt.Errorf("invalid peer ID. Error: [%w]", err)
```
Comment (Jorropo) [inline client/putipns.go:31]:
```suggestion
		return nil, fmt.Errorf("invalid peer ID: %w", err)
```

File: client/store.go [hunk 5]
```diff
@@ -2,20 +2,41 @@ package client
...
 )
 
 var _ routing.ValueStore = &Client{}
 
 // PutValue adds value corresponding to given Key.
 func (c *Client) PutValue(ctx context.Context, key string, val []byte, opts ...routing.Option) error {
-	return c.PutIPNS(ctx, []byte(key), val)
+	ns, path, err := record.SplitKey(key)
+	if err != nil {
+		return fmt.Errorf("invalid key. Error: [%w]", err)
```
Comment (Jorropo) [inline client/store.go:18]:
```suggestion
		return fmt.Errorf("invalid key: %w", err)
```

File: client/store.go [hunk 6]
```diff
@@ -2,20 +2,41 @@ package client
...
+
+	return c.PutIPNS(ctx, []byte(path), val)
 }
 
 // GetValue searches for the value corresponding to given Key.
 func (c *Client) GetValue(ctx context.Context, key string, opts ...routing.Option) ([]byte, error) {
-	return c.GetIPNS(ctx, []byte(key))
+	ns, path, err := record.SplitKey(key)
+	if err != nil {
+		return nil, fmt.Errorf("invalid key. Error: [%w]", err)
```
Comment (Jorropo) [inline client/store.go:32]:
```suggestion
		return nil, fmt.Errorf("invalid key: %w", err)
```

File: client/store.go [hunk 7]
```diff
@@ -29,7 +50,16 @@ func (c *Client) GetValue(ctx context.Context, key string, opts ...routing.Optio
 // Implementations of this methods won't return ErrNotFound. When a value
 // couldn't be found, the channel will get closed without passing any results
 func (c *Client) SearchValue(ctx context.Context, key string, opts ...routing.Option) (<-chan []byte, error) {
-	resChan, err := c.GetIPNSAsync(ctx, []byte(key))
+	ns, path, err := record.SplitKey(key)
+	if err != nil {
+		return nil, fmt.Errorf("invalid key. Error: [%w]", err)
```
Comment (Jorropo) [inline client/store.go:55]:
```suggestion
		return nil, fmt.Errorf("invalid key: %w", err)
```

**Review by Jorropo — APPROVED** [review]
LGTM

**Review by Jorropo — APPROVED** [review]
LGTM

**Pull request merged**